### DISCUSSION TITLE
Idea about plugins in supabase?

### DISCUSSION THREAD

**Discussion by Perfecto-always** [supabase/supabase#7582] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.

The taught came to my mind when I was thinking about storing images in supabase, wouldn't it be nice if their was something like a plugin to compress images before storing it.

## Additional context

As supabase says it's an open source alternative to firebase, one of its cool feature(at least to me) is it's plugins. Which in some cases serves as a complete replacement rather than creating their own backend.

I was wondering if supabase team is planning for any plugins like system maybe in the near future?

**Comment by eli5zen** [comment]
Hey @Perfecto-always , what would you suggest the contexts the plugins would run in should be?

As capabilities are different and depend on the host creating the context.

Different hosts/contexts I had in mind:

## postgres-meta - Injection during the API generation process

Another example of a context would the [postgres-meta](https://github.com/supabase/postgres-meta) server. Plugins would allow (for example) to exclude some tables/functions/views from the generated API. It can be a solution for issues similar to #7542 and #7563. The plugin would probably be executed somewhere around [here](https://github.com/supabase/postgres-meta/blob/bbe42502febc205467a142bf811014fd4663b4e0/src/server/templates/typescript.ts#L4).

## Realtime Server - extend the real-time functionality

In the [realtime](https://github.com/supabase/realtime) feature family, I guess the Erlang VM would be another example of a context. Examples for plugins would be different failover strategies and extensions of the real-time "language syntax".

These are some of my thoughts :)

**Reply by Perfecto-always** [reply]
I don't really have any details on how we can achieve this was more of a question to know are there any plans for it appearing in future implementation.(I'm sorry if I misunderstood your question).
Some scenarios I would *like* to use plugins would be for image compression, stripe payments, invoice mailings.

**Comment by kiwicopple** [comment]
A lot of great ideas here! Thanks for bringing this up. We will definitely think of "plugins" in the future, but for now we mostly want to collaborate with others first: https://supabase.com/partners/integrations

We'll put a but of work into the integrations platform, and then we will probably fill any gaps which our users need.

You can keep this thread for ideas for the future - ideally you create a single comment (and justification) per idea, so that others can upvote their favourites 👍

**Comment by landsman** [comment]
I'm interested in such a thing. I would love to write my own plugin for email sending, over my custom backend: https://github.com/orgs/supabase/discussions/18488
